lib/rack/query_parser.rb

```diff
@@ -207,8 +216,8 @@ def params_hash_has_key?(hash, key)
       true
     end
 
-    def unescape(s)
-      Utils.unescape(s)
+    def unescape(string, encoding = Encoding::UTF_8)
+      URI.decode_www_form_component(string, encoding)
```

Comment:
This seems unrelated to the change being made.  I'm not necessary opposed to this, but it shouldn't be part of this PR.  It can be a separate PR with a reason given for the change.  Ditto for the `uri` require at the top.

Reply:
I couldn't make the separate tests pass without it.

Reply:
Fix for this is simple, submitted in #2055.  However, considering this was backported without any discussion, I'm not sure what we want to do here.